Types/HttpQuery.php

```diff
@@ -35,21 +35,43 @@ class HttpQuery {
      * clause of the parsed sql tokens
      *
      * @param  array $tokens
+     * @param  array $options
      * @return string|null
      *
      * @SuppressWarnings("PHPMD.StaticAccess")
      */
-    public static function create(array $tokens) {
+    public static function create(array $tokens, array $options = []) {
         HashMap::assert($tokens, 'tokens');
 
         $operation = SqlOperation::create($tokens);
-        if ($operation !== SqlOperations::SELECT || empty($tokens['WHERE'])) return null;
+        if ($operation !== SqlOperations::SELECT) return null;
 
         $tableAlias = Table::alias($tokens);
-        $query      = array_reduce($tokens['WHERE'], function($query, $token) use ($tableAlias) {
-            return $query . str_replace('"', '', str_replace('OR', '|', str_replace('AND', '&', str_replace($tableAlias . '.', '', $token['base_expr']))));
-        });
+        
+        $query = '';
+        
+        if(isset($tokens['WHERE'])) {
```

Comment:
Try to avoid branching. We could use early return in this case.

```php
    if(!isset($tokens['WHERE'])) return preg_replace('/' . Identifier::column($tokens, new MetaData()) . '\=\d*&*/', '', '');

    $query = array_reduce($tokens['WHERE'], function($query, $token) use ($tableAlias) {
        return $query . str_replace('"', '', str_replace('OR', '|', str_replace('AND', '&', str_replace($tableAlias . '.', '', $token['base_expr']))));
    });

    if(!isset($options['pagination_as_query']) || !$options['pagination_as_query']) return preg_replace('/' . Identifier::column($tokens, new MetaData()) . '\=\d*&*/', '', $query);

    $perPageParam = isset($options['per_page_param']) ? $options['per_page_param'] : PaginationQuery::DEFAULT_PER_PAGE_PARAM;
    $pageParam    = isset($options['page_param']) ? $options['page_param'] : PaginationQuery::DEFAULT_PAGE_PARAM;

    $paginationParameters = PaginationQuery::create($tokens, $perPageParam, $pageParam);

    if(!$paginationParameters) return preg_replace('/' . Identifier::column($tokens, new MetaData()) . '\=\d*&*/', '', $query);

    return preg_replace('/' . Identifier::column($tokens, new MetaData()) . '\=\d*&*/', '', $query . http_build_query($paginationParameters));
```

This is just a note how the code could look like to give you an idea of what I'm talking about. I'm pretty sure I've made some mistakes. And you could still improve it (last 2 returns could be one by using the ternary operator).